src/apify/scrapy/cache.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import annotations
+
+import gzip
+import io
+import pickle
+import struct
+from logging import getLogger
+from time import time
+from typing import TYPE_CHECKING
+
+from scrapy.http.headers import Headers
+from scrapy.responsetypes import responsetypes
+
+from ._async_thread import AsyncThread
+from apify import Configuration
+from apify.apify_storage_client import ApifyStorageClient
+from apify.storages import KeyValueStore
+
+if TYPE_CHECKING:
+    from scrapy import Request, Spider
+    from scrapy.http.response import Response
+    from scrapy.settings import BaseSettings
+    from scrapy.utils.request import RequestFingerprinterProtocol
+
+logger = getLogger(__name__)
+
+
+class ApifyCacheStorage:
+    """A Scrapy cache storage that uses the Apify `KeyValueStore` to store responses.
+
+    This cache storage requires the asyncio Twisted reactor to be installed.
+    """
+
+    def __init__(self, settings: BaseSettings) -> None:
+        self.expiration_max_items = 100
+        self.expiration_secs: int = settings.getint('HTTPCACHE_EXPIRATION_SECS')
+        self.spider: Spider | None = None
+        self._kv: KeyValueStore | None = None
+        self._fingerprinter: RequestFingerprinterProtocol | None = None
+        self._async_thread: AsyncThread | None = None
+
+    def open_spider(self, spider: Spider) -> None:
+        """Open the cache storage for a spider."""
+        logger.debug('Using Apify key value cache storage', extra={'spider': spider})
+        self.spider = spider
+        self._fingerprinter = spider.crawler.request_fingerprinter
+        kv_name = f'httpcache-{spider.name}'
+
+        async def open_kv() -> KeyValueStore:
```

Comment:
```suggestion
        async def open_kvs() -> KeyValueStore:
```

we use "KVS" shortcut for key-value store.

Reply:
Addressed in 3909758